src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioDocumentTrackerFactoryFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    [Shared]
+    [ExportLanguageServiceFactory(typeof(VisualStudioDocumentTrackerFactory), RazorLanguage.Name, ServiceLayer.Default)]
+    internal class DefaultVisualStudioDocumentTrackerFactoryFactory : ILanguageServiceFactory
+    {
+        private readonly TextBufferProjectService _projectService;
+        private readonly ITextDocumentFactoryService _textDocumentFactory;
+        private readonly ImportDocumentManager _importDocumentManager;
+
+        [ImportingConstructor]
+        public DefaultVisualStudioDocumentTrackerFactoryFactory(
+            TextBufferProjectService projectService,
```

Comment:
@ajaybhargavb all but the text document factory will go away here once we swap to Roslyn services.